### PR TITLE
Exchange version update[v2] - currencycom

### PR DESCRIPTION
**Pull request by ttodua**

### CODE REVIEW

File: js/currencycom.js [hunk 1]
```diff
@@ -142,12 +141,22 @@ module.exports = class currencycom extends Exchange {
...
+                },
+                'private': {
+                    'get': [
+                        'v1/currencies',
+                        'v2/currencies',
+                        'v1/leverageSettings',
+                        'v1/openOrders',
+                        'v1/tradingPositions',
+                        'v1/account',
+                        'v1/myTrades',
```
Comment (ttodua) [inline js/currencycom.js:155]: oh, how I missed this. probably when cutting 'currencies' i've forgot to move other private endpoints from public. silly mistake, sorry.

File: js/currencycom.js [hunk 2]
```diff
@@ -830,7 +745,7 @@ module.exports = class currencycom extends Exchange {
         const market = this.market (symbol);
         const request = {
             'symbol': market['id'],
-            'interval': timeframe,
+            'interval': this.timeframes[timeframe],
```
Comment (ttodua) [inline js/currencycom.js:744]: this one too, was testing with direct loop, and missed to revert back.

File: js/currencycom.js [hunk 3]
```diff
@@ -91,10 +91,6 @@ module.exports = class currencycom extends Exchange {
                 'fetchTransfers': undefined,
                 'fetchWithdrawal': undefined,
                 'fetchWithdrawals': undefined,
-                'loadLeverageBrackets': undefined,
-                'loadMarkets': true,
-                'privateAPI': true,
-                'publicAPI': true,
```
Comment (ttodua) [inline js/currencycom.js]: Do we remove `privateAPI/publicAPI` keys from derived classes from now and onwards?

File: js/currencycom.js [hunk 4]
```diff
@@ -269,188 +266,104 @@ module.exports = class currencycom extends Exchange {
...
-        //          maxSLGap: 50,             // present in 'LEVERAGE' pairs
-        //          minSLGap: 1,              // present in 'LEVERAGE' pairs
-        //          maxTPGap: 50,             // present in 'LEVERAGE' pairs
-        //          minTPGap: 0.5,            // present in 'LEVERAGE' pairs
-        //          assetType: "CRYPTOCURRENCY",
-        //        },
-        //       ...
-        //   ]
-        // }
+        const response = await this.publicGetV1ExchangeInfo (params);
```
Comment (ttodua) [inline js/currencycom.js:269]: so this was also reverted to v1 (which returns incorrect data, based i.e. 30 seconds+ earlier) ?

File: js/currencycom.js [hunk 5]
```diff
@@ -948,15 +863,15 @@ module.exports = class currencycom extends Exchange {
...
+        //             "p":"0.02476",
+        //             "q":"0.0",
         //             "T":1591001423382,
-        //             "m":false        // Was the buyer the maker
+        //             "m":false
         //         }
         //     ]
         //
-        return this.parseTrades (response, market, since, limit, params);
+        return this.parseTrades (response, market, since, limit);
```
Comment (ttodua) [inline js/currencycom.js:870]: isn't it better to include 'params' too ?

**Comment by ttodua** [comment]
just resorted the api pathes, and adpoted your last changes.

File: js/currencycom.js [hunk 6]
```diff
@@ -269,188 +266,104 @@ module.exports = class currencycom extends Exchange {
...
-                    'deposit': {
-                        'min': this.safeNumber (currency, 'minDeposit'),
-                        'max': undefined,
-                    },
-                },
-            };
-        }
-        return result;
+        return this.safeInteger (response, 'serverTime');
     }
```
Comment (ttodua) [inline js/currencycom.js:267]: Why fetchCurrencies needed to be removed?

File: js/currencycom.js (hunk 5 above)
Comment (kroitor) [reply js/currencycom.js]: Not in this case – these params are for extending the resulting trades, so we don't want to extend all trades with any possible params a user might pass to this unified method.

File: js/currencycom.js (hunk 4 above)
Comment (kroitor) [reply js/currencycom.js:269]: It was temporarily reverted for a merge. We will restore this in consequent PRs.

File: js/currencycom.js (hunk 3 above)
Comment (kroitor) [reply js/currencycom.js]: They should be filled-in upon building.

File: js/currencycom.js (hunk 6 above)
Comment (kroitor) [reply js/currencycom.js:267]: I just wanted to make this PR less intrusive to merge it faster.

File: js/currencycom.js [hunk 7]
```diff
@@ -23,36 +23,86 @@ module.exports = class currencycom extends Exchange {
             'has': {
                 'CORS': undefined,
                 'spot': true,
-                'margin': undefined, // has but not fully implemented
-                'swap': false,
+                'margin': true,
+                'swap': true,
```
Comment (samgermain) [inline js/currencycom.js:27]: @ttodua I think this should still be false, I don't think currencycom has `swap` trading. Also, `swap` shouldn't be set to true until all the methods have swap functionality enabled, so until you can create a swap order, fetch the swap markets ... and from looking at `fetchMarkets` it looks like `swap` is always set to `false`

Comment (ttodua) [reply js/currencycom.js:27]: currencycom has swap yes, it's not margin trading, but `LEVERAGE` (they call so) which is swap as we checked it in the past. However, I'll double check the matter per your suggestion.

Comment (ttodua) [reply js/currencycom.js:27]: We had some discussion about it, and currencycom has swap: https://discord.com/channels/690203284119617602/921046068555313202/944277454011396136

Comment (samgermain) [reply js/currencycom.js:27]: > We had some discussion about it, and currencycom has swap: https://discord.com/channels/690203284119617602/921046068555313202/944277454011396136

it still shouldn't be set to true though until all the methods work for swap trading

Comment (ttodua) [reply js/currencycom.js:27]: I am not sure if that should be - if some methods are supported from swap, then we set 'true' and 'false' goes to specifically those methods that are not supported.  @kroitor please mention what is correct thing in such case.

Comment (samgermain) [reply js/currencycom.js:27]: I mean the methods like `createOrder` and stuff, if `swap` is set to true, the `createOrder`, `fetchOrder` ... should all work for swap

**Pull request merged**